client/components/data/query-purchase-cancellation-offers/index.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCancellationOffers } from 'calypso/state/cancellation-offers/actions';
+import isFetchingCancellationOffers from 'calypso/state/cancellation-offers/selectors/is-fetching-cancellation-offers';
+
+interface OwnProps {
+	siteId: number;
+	purchaseId: number;
+}
+
+const QueryPurchaseCancellationOffers = ( { siteId, purchaseId }: OwnProps ) => {
+	const dispatch = useDispatch();
+
+	const fetchingCancellationOffers = useSelector( ( state ) =>
+		isFetchingCancellationOffers( state, purchaseId )
+	);
+
+	useEffect( () => {
+		if ( siteId && fetchingCancellationOffers === null ) {
```

Comment:
Is the reason for checking null (instead of a falsy value) to make sure the endpoint is called only once during the user session?

Reply:
That's correct, this was the approach that I took to know that the query had never been run.
I tried having `isFetchingCancellationOffers` return only a boolean, but ran into the trouble of how to know whether or not the query had been attempted before in a simple way.

Reply:
Understood. Any chance this could lead to the redux state having stale data at some point? Risk seems low to me, but what's your take on this?